pkg/beholder/noop.go

```diff
@@ -0,0 +1,139 @@
+
+	// Meter
+	meterProvider := otelmetricnoop.NewMeterProvider()
+	meter := meterProvider.Meter(defaultPackageName)
+
+	// MessageEmitter
+	messageEmitter := noopMessageEmitter{}
+
+	client := Client{cfg, logger, tracer, meter, messageEmitter, loggerProvider, tracerProvider, meterProvider, loggerProvider, noopOnClose}
+
+	return &client
+}
+
+// NewStdoutClient creates a new Client with stdout exporters
+// Use for testing and debugging
+// Also this client is used as a noop client when otel exporter is not initialized properly
+func NewStdoutClient(opts ...StddutClientOption) (*Client, error) {
+	cfg := DefaultStdoutClientConfig()
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+	// Logger
+	loggerExporter, err := stdoutlog.New(
+		append([]stdoutlog.Option{stdoutlog.WithoutTimestamps()}, cfg.LogOptions...)...,
+	)
+	if err != nil {
+		return NewNoopClient(), err
+	}
+	loggerProvider := sdklog.NewLoggerProvider(sdklog.WithProcessor(sdklog.NewSimpleProcessor(loggerExporter)))
+	logger := loggerProvider.Logger(defaultPackageName)
+	otel.SetErrorHandler(otel.ErrorHandlerFunc(func(err error) {
+		fmt.Printf("OTel error %s", err)
+	}))
+
+	// Tracer
+	traceExporter, err := stdouttrace.New(cfg.TraceOptions...)
+	if err != nil {
+		return NewNoopClient(), err
+	}
+
+	tracerProvider := sdktrace.NewTracerProvider(sdktrace.WithSpanProcessor(
+		sdktrace.NewSimpleSpanProcessor(traceExporter),
+	))
+	tracer := tracerProvider.Tracer(defaultPackageName)
+
+	// Meter
+	metricExporter, err := stdoutmetric.New(cfg.MetricOptions...)
+	if err != nil {
+		return NewNoopClient(), err
+	}
+	meterProvider := sdkmetric.NewMeterProvider(
+		sdkmetric.WithReader(
+			sdkmetric.NewPeriodicReader(
+				metricExporter,
+				sdkmetric.WithInterval(100*time.Millisecond), // Default is 10s
+			)),
+	)
+	meter := meterProvider.Meter(defaultPackageName)
+
+	// MessageEmitter
+	emitter := messageEmitter{messageLogger: logger}
+
+	onClose := func() (err error) {
+		for _, provider := range []shutdowner{loggerProvider, tracerProvider, meterProvider} {
+			err = errors.Join(err, provider.Shutdown(context.Background()))
+		}
+		return
+	}
+
+	client := Client{cfg.Config, logger, tracer, meter, emitter, loggerProvider, tracerProvider, meterProvider, loggerProvider, onClose}
+
+	return &client, nil
+}
+
+type noopMessageEmitter struct{}
+
+func (noopMessageEmitter) Emit(ctx context.Context, body []byte, attrKVs ...any) error {
+	return nil
+}
+func (noopMessageEmitter) EmitMessage(ctx context.Context, message Message) error {
+	return nil
+}
+
+func noopOnClose() error {
+	return nil
+}
+
+type StddutClientOption func(*StdoutClientConfig)
+
+type StdoutClientConfig struct {
+	Config
+	LogOptions    []stdoutlog.Option
+	TraceOptions  []stdouttrace.Option
+	MetricOptions []stdoutmetric.Option
+}
+
+func DefaultStdoutClientConfig() StdoutClientConfig {
+	return StdoutClientConfig{
+		Config: DefaultConfig(),
+	}
+}
+
+func WithWriter(w io.Writer) StddutClientOption {
```

Comment:
For consistency, we should rename `NewStdoutClient` since it can be configured to write to other places.
I also wonder if functional options are the right fit for this. What about starting with something simple, like:
```go
func NewStdoutClient() (*Client, error) {
	return NewWriterClient(os.Stdout)
}
func NewWriterClient(w io.Writer) (*Client, error) {
	...
}
```

Reply:
And then `StdoutClientConfig` could be `WriterClientConfig` but also doesn't seem to need to be exported, does it?

Reply:
> And then StdoutClientConfig could be WriterClientConfig but also doesn't seem to need to be exported, does it?

It could be very handy for testing: use StdoutClient instead of OtlpClient to verify that otel instrumentation emits correct data from the code.

Reply:
But the API in its current form doesn't really make that config struct very usable, because a caller would have to write their own functional options to access it. If we want to expose that config, why not do it directly like the standard constructor?

Reply:
I agree with this point `And then StdoutClientConfig could be WriterClientConfig` 

See https://github.com/smartcontractkit/chainlink-common/pull/694/commits/3504324de7b444664cd40e5ed1ab70f42a6c4480

Reply:
> But the API in its current form doesn't really make that config struct very usable, because a caller would have to write their own functional options to access it. If we want to expose that config, why not do it directly like the standard constructor?

Right, makes sense, I misunderstood original suggestion 

https://github.com/smartcontractkit/chainlink-common/pull/694/commits/2187c017d32052632b16fe25f40034fa0d5a4ae9